### PR TITLE
HPCC-18565 Support roxie log capture via SOAP and WSECL

### PR DESCRIPTION
**Pull request by afishbeck**
Currently roxie allows a caller to capture the log messages associated
with their query when the query is run using roxie's native
protocol.

Add support for log capture to the roxie SOAP and JSON protocols.

Also add support for using the feature to WsECL (WsECL support
existed pre-oss, when WsECL called roxie via its native protocol).

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by afishbeck** [comment]
@richardkchapman please see if this is still ok for a point release.  The change doesn't just affect WsECL.  Roxie only supported getting log information via its native protocol.  This change adds roxie support for capturing log info via SOAP and JSON as well.

File: esp/services/ws_ecl/ws_ecl_service.cpp [hunk 1]
```diff
@@ -558,7 +558,16 @@ static void buildReqXml(StringArray& parentTypes, IXmlType* type, StringBuffer&
...
             if (attrval)
-                appendXMLAttr(out, attr->queryName(), attrval);
+                appendXMLAttr(out, attr->queryName(), attrval, nullptr, true);
+        }
+        if (flags & REQSF_ROOT)
+        {
+            bool log = reqTree->getPropBool("@log", false);
+            if (log)
+                appendXMLAttr(out, "log", "true", nullptr, true);
+            const char *tracelevel = reqTree->queryProp("@traceLevel");
```
Comment (richardkchapman) [inline esp/services/ws_ecl/ws_ecl_service.cpp]: Why not getPropInt (as it is above) ?

Comment (afishbeck) [reply esp/services/ws_ecl/ws_ecl_service.cpp]: I'll change it.  Was necessary for JSON to get the type formatting right, but would be better here as well.

**Comment by afishbeck** [comment]
@richardkchapman updated to use getPropInt().

**Comment by richardkchapman** [comment]
@ghalliday Can you review please - I'm nervous for a opint release, and would like a second pair of eyes/opinion

**Pull request merged**